runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -264,14 +266,117 @@ export default class ExcelActionHandler extends StatefulActionHandler {
      * and we expect it does not exist, or if it does not exist and we expect it to exist.
      */
     private checkWorksheet(worksheet: string, shouldExist: boolean): void {
-        if (
-            (shouldExist && !this.checkIfWorksheetExist(worksheet)) ||
-            (!shouldExist && (worksheet.trim() === '' || this.checkIfWorksheetExist(worksheet)))
-        ) {
-            throw new Error(ExcelErrorMessage.worksheetIncorrectInput(shouldExist));
+
+if (
+    (shouldExist && !this.checkIfWorksheetExist(worksheet)) ||
+    (!shouldExist && (worksheet.trim() === '' || this.checkIfWorksheetExist(worksheet)))
+) {
+    throw new Error(ExcelErrorMessage.worksheetIncorrectInput(shouldExist));
+}
+
+    }
+    async runMacro(input: ExcelRunMacroInput) {
+
+        if(input.functionParams == undefined){
+            return this.session.Run(input.macro);
+        }
```

Comment:
handle empty list here also. Maybe currently our frontend does not allow it to happen but it should be ready to handle it in the future